chainstate/tx-verifier/src/transaction_verifier/mod.rs

```diff
@@ -502,8 +513,14 @@ where
             // TODO: see if a different treatment should be done for different output purposes
             // TODO: ensure that signature verification is tested in the test-suite, they seem to be tested only internally
             match utxo.output().purpose().destination() {
-                Some(d) => verify_signature(self.chain_config.as_ref(), d, tx, input_idx)
-                    .map_err(ConnectTransactionError::SignatureVerificationFailed)?,
+                Some(d) => verify_signature(
+                    self.chain_config.as_ref(),
+                    d,
+                    tx,
+                    &inputs_utxos.iter().collect::<Vec<_>>(),
```

Comment:
then here simply `&inputs` can be passed

Reply:
That won't work. This is a Vec of values. We want a slice of refs.

Reply:
I mean that would work in combination with my prev comment

Reply:
It was like that before. I changed it to slice of refs. I'll resolve this and move that discussion there.